src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -668,6 +672,26 @@ public void onOnline(Computer c, TaskListener listener) throws IOException, Inte
                         "Could not install root log handler on node: " + c.getName(), e);
             }
         }
+
+        @Override
+        public void onOffline(@Nonnull Computer c, @CheckForNull OfflineCause cause) {
+            if (Boolean.getBoolean(SupportPlugin.class.getCanonicalName() + ".generateOnOfflineNode")) {
+                boolean locked = false;
+                try {
+                    locked = lock.tryLock();
+                    if (locked) {
+                        PeriodicWork.all().get(PeriodicWorkImpl.class).doRun();
```

Comment:
It would be better to log something like "Node xxx has gone offline, so a support bundle was generated. Please check $JENKINS_HOME/support for a bundle generated at this time.